### PR TITLE
Add InteractionsModel for systematic model training an serialization

### PR DESCRIPTION
**Pull request by rfl-urbaniak**
Introduce `InteractionsModel` class and a test thereof. Objects in this class:
- load and prepare appropriate data for inference (still to be revised once the new transformation method lands)
- train the interactions model on a given combination of data and forward shift
- serialize and save the resulting guide

A example of usage and downstream functioning is in the updated version of `interaction_cities_prototype.ipynb`.

### CODE REVIEW

File: cities/modeling/model_interactions.py [hunk 1]
```diff
@@ -1,7 +1,89 @@
...
+        self.guide = train_interactions_model(
+            model=self.model_conditioned,
+            model_args=self.model_args,
+            num_iterations=self.num_iterations,
+            plot_loss=self.plot_loss,
+        )
+
+    def save_guide(self):
+        guide_name = (
+            f"{self.intervention_dataset}_{self.outcome_dataset}_{self.forward_shift}"
```
Comment (azane) [inline cities/modeling/model_interactions.py:76]: maybe explain how these three things uniquely identify this interactions model

File: tests/test_inference.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from cities.modeling.model_interactions import InteractionsModel
+
+
+# TODO this needs to be parametrized by outcome datasets,
+# intervention datasets and forward shifts
+def test_InteractionsModel():
```
Comment (azane) [inline tests/test_inference.py:6]: sometimes I'll call these "..._runtime_test" explicitly just so I don't think it's testing correctness.

**Review by azane — APPROVED** [review]
lgtm, maybe we can look more about full counterfactual angles too.

**Review by riadas — APPROVED** [review]
LGTM

**Pull request merged**